File: frontend/src/api/emailSettings.ts
import { api } from './client';
import type { NotificationSeverity } from './models';

export type SmtpSecurity = 'None' | 'StartTls' | 'Auto' | 'SslOnConnect';

export interface EmailSettings {
  enabled: boolean;
  smtpHost: string;
  smtpPort: number;
  security: SmtpSecurity;
  username: string | null;
  passwordSet: boolean;
  fromAddress: string;
  fromName: string;
  appBaseUrl: string | null;
  minSeverity: NotificationSeverity;
}

export interface UpdateEmailSettings {
  enabled: boolean;
  smtpHost: string;
  smtpPort: number;
  security: SmtpSecurity;
  username: string | null;
  /** Write-only: omit/empty keeps the stored password. */
  password: string | null;
  fromAddress: string;
  fromName: string;
  appBaseUrl: string | null;
  minSeverity: NotificationSeverity;
}

export const emailSettingsApi = {
  /** GET returns 204 (undefined) when never configured. */
  get: () => api.get<EmailSettings | undefined>(`/api/email-settings`),
  update: (body: UpdateEmailSettings) => api.put<EmailSettings>(`/api/email-settings`, body),
  sendTest: () => api.post<void>(`/api/email-settings/test`),
};

File: Proxytrace.Application/Notifications/EmailSettings.cs
using Proxytrace.Domain.Notification;

namespace Proxytrace.Application.Notifications;

/// <summary>How the SMTP connection is secured (maps to MailKit's <c>SecureSocketOptions</c>).</summary>
public enum SmtpSecurity
{
    None,
    StartTls,
    Auto,
    SslOnConnect,
}

/// <summary>
/// Operator-configured outgoing-email settings. A single instance per installation. <see cref="Password"/>
/// is the plaintext SMTP password in memory; it is encrypted at rest by the store via
/// <see cref="Proxytrace.Application.Security.ISecretProtector"/>.
/// </summary>
public sealed record EmailSettings(
    bool Enabled,
    string SmtpHost,
    int SmtpPort,
    SmtpSecurity Security,
    string? Username,
    string? Password,
    string FromAddress,
    string FromName,
    string? AppBaseUrl,
    NotificationSeverity MinSeverity);
